Remove the dropzone global attribute.

pwnall opened the pull request:
dropzone failed to get traction among browser implementers. Having it in the specification is confusing to Web developers who may attempt to use it, only to discover that it is not supported.

Safari and Chrome implemented a prefixed version, webkitdropzone. The prefixed version is going away from Chrome [1]. Other browser vendors have no objections to the attribute getting removed from the spec [2].

[1] https://www.chromestatus.com/feature/5718005866561536
[2] https://github.com/whatwg/html/issues/2331

This fixes #2331.

pwnall committed:
Remove the dropzone global attribute.

dropzone failed to get traction among browser implementers. Having it in
the specification is confusing to Web developers who may attempt to use
it, only to discover that it is not supported.

Safari and Chrome implemented a prefixed version, webkitdropzone. The
prefixed version is going away from Chrome [1]. Other browser vendors
have no objections to the attribute getting removed from the spec [2].

[1] https://www.chromestatus.com/feature/5718005866561536
[2] https://github.com/whatwg/html/issues/2331

domenic commented:
Thanks for doing this!

It looks like there are some build errors; would you mind fixing so we can review more easily?

pwnall commented:
@domenic On it! I'm still trying to figure out what the errors mean, but I will definitely not give up until I get a green build.

pwnall commented:
@domenic Can you please take another look?

I was looking at the line of the first error, until I realized that the 2nd line number is smaller than the first. 😅

domenic commented:
Build is passing now, so I'll be able to compile locally and do a check :). It's the weekend at this point though so I'll take a look Monday!

pwnall commented:
Thank you very much!

domenic commented on source:
```diff
@@ -74264,33 +74262,45 @@ body { display:none }
-  give (e.g. "<code data-x="">move</code>" to indicate that the data will be moved).</p>
+  <ol>
+
+    <li>The <code data-x="event-dnd-dragenter">dragenter</code> event handler must report
```
It might be helpful to add something like "by canceling the event"

uzkjzjjz replied:
@uzkjzjjz

domenic commented on source:
```diff
@@ -74264,33 +74262,45 @@ body { display:none }
+  function dragOverHandler(event) {
+    event.dataTransfer.dropEffect = 'move';
+  }
   function dropHandler(event) {
```
The example below this one also needs updating, to stay in sync.

pwnall replied:
I tried out the full example, and (assuming I got it right) the code fragment below doesn't need upgrading.

domenic replied:
Well, the issue is that the code fragment below is supposed to be a modification of the above one (which you modified) to simply add dragend handling. So since you added dragenter/dragover handling to the first fragment, the second fragment should get updated to match.

pwnall replied:
GitHub seems to fail at expanding the diff (perhaps because `source` is a large file), so you'll have to use a local checkout to look at the lines I'm referencing here. Sorry about that!

IIUC, you're referring to lines 74331-74345. They're an update of lines 74244-74261, which I haven't changed. Can you please explain me what I got wrong? Also, thank you for bearing with me!

domenic replied:
No problem!

The issue is that lines 74331-74345 are meant to be exactly the same example as lines 74289-74322, but with the addition of also adding a dragend handler. That is, consider the sentence between the two examples, annotated:

> For our example here [i.e., the example on lines 74289-74322], that means updating the original markup [shown in those lines] to handle that [dragend] event, [as shown below by modifying the original markup but also adding some dragend-specific stuff].

This it's important that lines 74331-74345 match lines 74289-74322, except:
 - They should add dragend handling
 - They can use "... as before ..." to avoid repetition. (I might just do `// dragEnterHandler, dragOverHandler, and dropHandler as before` instead of the current `function dragStartHandler(event) { // <em>...as before...</em> }`)

domenic replied:
Clarified offline: I was confused. As @pwnall says, lines 74331-74345 are an update of lines 74244-74261, *not* an update of lines 74289-74322 like I thought.

domenic reviewed (changes requested):
A few minor things to fix, but overall looking great!

pwnall commented on source:
```diff
@@ -74264,33 +74262,45 @@ body { display:none }
-  give (e.g. "<code data-x="">move</code>" to indicate that the data will be moved).</p>
+  <ol>
+
+    <li>The <code data-x="event-dnd-dragenter">dragenter</code> event handler must report
```
Done. Thank you for the suggestion! Thinking back to the first time I coded up drag-and-drop, more clarity here is super-helpful.

pwnall commented on source:
```diff
@@ -74264,33 +74262,45 @@ body { display:none }
+    whether or not the drop target is potentially willing to accept the drop.</li>
+
+    <li>The <code data-x="event-dnd-dragover">dragover</code> event handler must specify what
+    feedback will be shown to the user.</li>
```
I tried a draft that simply removed the "must". WDYT?

domenic replied:
Looks good!

pwnall commented:
@domenic Thank you very much for your thoughtful feedback! PTAL?

pwnall commented:
@domenic Thank you very much for helping me with my first contribution!

domenic commented:
This LGTM! I forgot to ask though, we are trying to get web platform tests before merging any changes. I'm happy to help with that if you want, just let me know. Mostly it will be a matter of:

- deleting html/editing/dnd
- Removing any other tests that depend on dropzone; https://github.com/w3c/web-platform-tests/search?p=7&q=dropzone&utf8=%E2%9C%93 shows a few
- Adding a line to https://github.com/w3c/web-platform-tests/blob/master/html/dom/elements/elements-in-the-dom/historical.html

pwnall commented:
@domenic I started working on WPT.

I didn't remove everything with `dropzone` because I suspect that some other APIs (e.g. `DataTransfer.dropEffect`) might only be covered in tests that have `dropzone`. This was definitely the case in Blink. Should I ping this PR when I've removed everything?

domenic commented:
Hmm, interesting. How long do you think that would take? I don't think we should hold up merging this much longer. So I think it would be fine to merge this with even just a small test addition to https://github.com/w3c/web-platform-tests/blob/master/html/dom/elements/elements-in-the-dom/historical.html and an open issue on the web-platform-tests repository tracking the removal of the rest.

zcorpan commented:
`dropzone` has existed in the spec since 2010, so it seems reasonable to add it to the Obsolete section (with `id="the-dropzone-attribute"`).

Otherwise LGTM.

pwnall commented:
@zcorpan I added `dropzone` to the Obsolete section. Can you please take a look and check that I used appropriate text?

pwnall commented:
@domenic Thank you very much for your pragmatic advice!

I submitted https://github.com/w3c/web-platform-tests/pull/5052 for adding `dropzone` to the historical checks, and filed https://github.com/w3c/web-platform-tests/issues/5053

domenic committed:
Tweak position of the obsolete entry

Let's leave the massive "use CSS instead of all these presentational attributes" section at the bottom.

domenic commented:
It all looks good to me! I pushed a small tweak to move it before all the "use CSS instead" items. I'll merge after the CI completes!

pwnall commented:
Thank you very much for the feedback and for all the help & patience!

cklim483 commented:
> A few minor things to fix, but overall looking great!
